wsmaster/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/AgentService.java

```diff
@@ -0,0 +1,135 @@
+import static org.eclipse.che.api.agent.server.DtoConverter.asDto;
+
+/**
+ * Defines Agent REST API.
+ *
+ * @author Anatoliy Bazko
+ */
+@Api(value = "/agent", description = "Agent REST API")
+@Path("/agent")
+public class AgentService extends Service {
+
+    private final AgentRegistry agentRegistry;
+
+    @Context
+    private SecurityContext securityContext;
+
+    @Inject
+    public AgentService(AgentRegistry agentRegistry) {
+        this.agentRegistry = agentRegistry;
+    }
+
+    @POST
+    @Path("{name}")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Create a new agent", response = AgentDto.class)
+    @ApiResponses({@ApiResponse(code = 200, message = "The agent successfully created"),
+                   @ApiResponse(code = 404, message = "Agent not found"),
+                   @ApiResponse(code = 500, message = "Internal server error occurred")})
+    public Response create(@ApiParam("The agent name") @PathParam("name") String name) throws ServerException, NotFoundException {
+        try {
+            Agent agent = agentRegistry.createAgent(name);
+            return Response.status(Response.Status.OK).entity(asDto(agent)).build();
+        } catch (AgentNotFoundException e) {
+            throw new NotFoundException(e.getMessage());
+        } catch (AgentException e) {
+            throw new ServerException(e.getMessage());
+        }
+    }
+
+    @POST
+    @Path("{name}/{version}")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Create a new agent", response = AgentDto.class)
+    @ApiResponses({@ApiResponse(code = 200, message = "The agent successfully created"),
+                   @ApiResponse(code = 404, message = "Agent not found"),
+                   @ApiResponse(code = 500, message = "Internal server error occurred")})
+    public Response create(@ApiParam("The agent name") @PathParam("name") String name,
+                           @ApiParam("The agent version") @PathParam("version") String version) throws ServerException, NotFoundException {
+
+        try {
+            Agent agent = agentRegistry.createAgent(name, version);
+            return Response.status(Response.Status.OK).entity(asDto(agent)).build();
+        } catch (AgentNotFoundException e) {
+            throw new NotFoundException(e.getMessage());
+        } catch (AgentException e) {
+            throw new ServerException(e.getMessage());
+        }
+    }
+
+    @GET
+    @Path("versions")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Get a list of the available versions of the specific agent", response = List.class)
+    @ApiResponses({@ApiResponse(code = 200, message = "OK"),
+                   @ApiResponse(code = 404, message = "Agent not found"),
+                   @ApiResponse(code = 500, message = "Internal server error occurred")})
+    public Response getVersions(@ApiParam("The agent name") @PathParam("name") String name) throws ServerException, NotFoundException {
```

Comment:
It seems that this method won't work as there is no `{name}` in `@Path("versions")`